### DISCUSSION TITLE
Best procedere to get phase information?

### DISCUSSION THREAD

**Discussion by ppetrecca** [MagneticResonanceImaging/MRIReco.jl#203] [General]
Hi, I'm using this package to reconstruct some 3D UTE scans. Both for debugging purposes and for clarity, it would be great to get my hands on the k-space directly after it has been reconstructed to cartesian. Also, and that's my main question, what is the proposed way to look at phase information? I've been doing 
```
        params[:reco] = "direct"
        Ireco = reconstruction(acq, params);
        myMatrix = collect(Ireco)
```
and then looking at `angle.(myMatrix)` but I was wondering whether that's the best way to do it or whether I'm missing something since julia is kind of uncharted territory for me.

**Comment by tknopp** [comment]
You can convert the raw data to `AcquisitionData` as can be seen here: https://github.com/MagneticResonanceImaging/MRIReco.jl/blob/master/MRIFiles/test/testBrukerFile.jl#L50
Afterwards, you can call `kData` or `kDataCart`, see https://github.com/MagneticResonanceImaging/MRIReco.jl/blob/master/MRIBase/src/Datatypes/AcqData.jl#L153C5-L153C14
to extract the k-space data.

Finally, yes, with `angle.` you get the phase information. But yours is already in image space.

**Comment by aTrotier** [comment]
If you want the kspace on the cartesian grid, perform a 3D fft onto the images.
